source/SemVerAnalyzer/Nuget/NugetClient.cs

```diff
@@ -21,7 +25,32 @@ public async Task<byte[]> GetAssemblyBytesFromPackage(string packageName, string
 			try
 			{
 				using var client = new HttpClient();
-				using var request = new HttpRequestMessage(HttpMethod.Get, Path.Join(_config.RepositoryUrl, packageName));
+				
+				using var feedRequest = new HttpRequestMessage(HttpMethod.Get, Path.Combine(_config.RepositoryUrl, "v3", "index.json"));
+				using var feedResponse = await client.SendAsync(feedRequest);
+				if (!feedResponse.IsSuccessStatusCode)
+				{
+					var content = await feedResponse.Content.ReadAsStringAsync();
+					comments.Add($"Error retrieving Nuget feed:\n{content}");
+					return null;
+				}
+
+				var feed = JsonSerializer.Deserialize<NugetPackageFeed>(await feedResponse.Content.ReadAsStringAsync());
+				var packageBaseAddress = feed.Resources.Single(r => r.Type.StartsWith("PackageBaseAddress")).Id;
+				
+				using var versionRequest = new HttpRequestMessage(HttpMethod.Get, Path.Combine(packageBaseAddress, packageName, "index.json"));
+				using var versionResponse = await client.SendAsync(versionRequest);
+				if (!versionResponse.IsSuccessStatusCode)
+				{
+					var content = await versionResponse.Content.ReadAsStringAsync();
+					comments.Add($"Error retrieving package versions:\n{content}");
+					return null;
+				}
+
+				var versions = JsonSerializer.Deserialize<VersionsFeed>(await versionResponse.Content.ReadAsStringAsync()).Versions;
+				var highestVersion = versions.OrderByDescending(v => v.MajorVersion()).ThenByDescending(v => v.MinorVersion()).ThenByDescending(v => v.PatchVersion()).First();
```

Comment:
I strongly recommend chopping this line (wrapping before each chained method) for readability.

Reply:
👍🏼